src/vs/platform/backup/common/backup.ts

```diff
@@ -20,10 +24,14 @@ export interface IBackupMainService {
 	isHotExitEnabled(): boolean;
 
 	getWorkspaceBackups(): IWorkspaceIdentifier[];
-	getFolderBackupPaths(): string[];
+	getFolderBackupPaths(): URI[];
 	getEmptyWindowBackupPaths(): string[];
 
 	registerWorkspaceBackupSync(workspace: IWorkspaceIdentifier, migrateFrom?: string): string;
-	registerFolderBackupSync(folderPath: string): string;
+	registerFolderBackupSync(folderUri: URI): string;
 	registerEmptyWindowBackupSync(backupFolder?: string): string;
+
+	unregisterWorkspaceBackupSync(workspace: IWorkspaceIdentifier): void;
```

Comment:
Why are these unregister functions needed now and not before? Aren't the workspaces dropped when the folder is opened again and there are no backups remaining?

Reply:
@aeschli Can you please address @Tyriar comments?

Reply:
There are tests that are removing folders. Previously the used an internal method. I decided it's cleaner to make these APIs public.
Alternative is to modify the tests.